File: thorough training/arrays/most_frequent.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $k
     * @return Integer[]
     */
    function topKFrequent($nums, $k) {
        $freq = [];

        foreach($nums as $num) {
            if(isset($freq[$num])) {
                $freq[$num]++;
            } else {
                $freq[$num] = 1;
            }
        } 

        $heap = new SplPriorityQueue;
        foreach($freq as $key => $count) {
            $heap->insert($key, $count);
        }

        $result = [];
        for($i = 0; $i < $k; $i++) {
            $result[] = $heap->extract();
        }

        return $result;

    }
}

File: thorough training/arrays/most_frequent.py
import heapq

class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        freq = {}

        for num in nums:
            freq[num] =  freq.get(num, 0) + 1

        heap = []

        for num, count in freq.items():
            heapq.heappush(heap, (-count, num))
        
        result = []
        for _ in range(k):
            count, num = heapq.heappop(heap)
            result.append(num)

        return result
        